### PR TITLE
Remove some warnings shown by GCC on Linux

### PR DESCRIPTION
**Pull request by chunga**
These patches mainly get rid of unused variables or static functions, but there are also some changes around conditional code.

### CODE REVIEW

File: src/driver/implementation/imap/imapdriver_cached.c [hunk 1]
```diff
@@ -656,10 +656,9 @@ static int imapdriver_cached_select_folder(mailsession * session, const char * m
   int r;
   char * quoted_mb;
   struct imap_cached_session_state_data * data;
-  mailsession * imap;
   char * old_mb;
   
-  imap = get_ancestor(session);
+  get_ancestor(session);
```
Comment (dinhviethoa) [inline src/driver/implementation/imap/imapdriver_cached.c]: You can probably remove this line altogether.

File: src/driver/implementation/imap/imapdriver_cached.c [hunk 2]
```diff
@@ -878,7 +876,7 @@ static int boostrap_cache(mailsession * session)
   chash * keys_uid;
   
   data = get_cached_data(session);
-  imap = get_imap_session(session);
+  get_imap_session(session);
```
Comment (dinhviethoa) [inline src/driver/implementation/imap/imapdriver_cached.c]: You can probably remove this line altogether.

File: src/driver/implementation/imap/imapdriver_tools.c [hunk 3]
```diff
@@ -3700,9 +3692,9 @@ imapdriver_get_cached_envelope(struct mail_cache_db * cache_db,
   
 #if 0
   imap_session = cached_session_get_ancestor(session);
-  imap = ((struct imap_session_state_data *) (imap_session->data))->session;
+  ((struct imap_session_state_data *) (imap_session->data))->session;
 #endif
-  imap = cached_session_get_imap_session(session);
+  cached_session_get_imap_session(session);
```
Comment (dinhviethoa) [inline src/driver/implementation/imap/imapdriver_tools.c]: You can remove this line.

File: src/driver/implementation/imap/imapdriver_tools.c [hunk 4]
```diff
@@ -3692,7 +3685,6 @@ imapdriver_get_cached_envelope(struct mail_cache_db * cache_db,
 #if 0
```
Comment (dinhviethoa) [inline src/driver/implementation/imap/imapdriver_tools.c]: Feel free to cleanup this #if 0.

File: src/driver/implementation/imap/imapdriver_tools.c [hunk 5]
```diff
@@ -3700,9 +3692,9 @@ imapdriver_get_cached_envelope(struct mail_cache_db * cache_db,
   
 #if 0
```
Comment (dinhviethoa) [inline src/driver/implementation/imap/imapdriver_tools.c]: same here, feel free to cleanup this #if 0.

File: src/driver/implementation/nntp/nntpdriver.c [hunk 6]
```diff
@@ -617,9 +617,8 @@ static int nntpdriver_append_message(mailsession * session,
 				     const char * message, size_t size)
 {
   int r;
-  struct nntp_session_state_data * data;
 
-  data = get_data(session);
+  get_data(session);
```
Comment (dinhviethoa) [inline src/driver/implementation/nntp/nntpdriver.c]: You can remove this line.

File: src/driver/implementation/pop3/pop3driver.c [hunk 7]
```diff
@@ -208,9 +208,8 @@ static int pop3driver_starttls(mailsession * session)
 {
   int r;
   mailpop3 * pop3;
-  struct pop3_session_state_data * data;
 
-  data = get_data(session);
+  get_data(session);
```
Comment (dinhviethoa) [inline src/driver/implementation/pop3/pop3driver.c]: You can remove this line.

File: src/driver/tools/mailthread.c [hunk 8]
```diff
@@ -373,14 +369,11 @@ static char * extract_subject(char * default_from,
     */
 
     if (len >= 5) {
-      size_t saved_begin;
-
-      saved_begin = begin;
       if (strncasecmp(subj + begin, "[fwd:", 5) == 0) {
 	begin += 5;
 	
 	if (subj[len - 1] != ']')
```
Comment (dinhviethoa) [inline src/driver/tools/mailthread.c]: Replace this line if if ( == ']') and remove the line with the single `;`.

File: src/engine/mailprivacy_gnupg.c [hunk 9]
```diff
@@ -2837,17 +2823,17 @@ static int mailprivacy_gnupg_init_lock_done = 0;
 #endif
 static chash * encryption_id_hash = NULL;
 
-static void mailprivacy_gnupg_init_lock(void)
```
Comment (dinhviethoa) [inline src/engine/mailprivacy_gnupg.c]: Why did you move those lines?

Comment (chunga) [inline src/engine/mailprivacy_gnupg.c]: GCC warned that mailprivacy_gnupg_init_lock() wasn't used anywhere, so I would have removed it; but then I saw that this code is executed `#elif (defined WIN32)`, so I thought "I can't test it under Windows, let's move the whole function under this `#elif`": one less warning under Linux, no change under Windows.

Comment (dinhviethoa) [inline src/engine/mailprivacy_gnupg.c]: I see. That part of the change looks good then!

**Comment by chunga** [comment]
Updated patches. I think I've addressed all of your comments.

File: src/driver/implementation/imap/imapdriver.c [hunk 10]
```diff
@@ -1119,9 +1119,8 @@ static int imapdriver_starttls(mailsession * session)
   clistiter * cur;
   int starttls;
   int capability_available;
-  struct imap_session_state_data * data;
   
-  data = get_data(session);
+  get_data(session);
```
Comment (dinhviethoa) [inline src/driver/implementation/imap/imapdriver.c]: Could you remove this line? Thanks.

File: src/engine/mailengine.c [hunk 11]
```diff
@@ -865,7 +863,7 @@ static int folder_connect(struct storage_ref_info * ref_info,
   if (r != MAIL_NO_ERROR)
     return r;
 
-  folder_ref_info = storage_get_folder_ref(ref_info, folder);
+  storage_get_folder_ref(ref_info, folder);
```
Comment (dinhviethoa) [inline src/engine/mailengine.c]: You can get rid of this line.

**Comment by dinhviethoa** [comment]
Two minor changes and then, it's good to be merged.

**Comment by chunga** [comment]
Updated. Sorry for being so sloppy and thanks for your comments.

**Comment by dinhviethoa** [comment]
Thanks a lot!
